tasks/preflight.yml

```diff
@@ -77,3 +79,64 @@
   register: prometheus_systemd_version
   tags:
     - skip_ansible_lint
+
+- block:
+  - name: Fail if docker labels are set but docker isn't
+    debug:
+      msg: "Labels are set, but the role isn't configured to install prometheus with docker"
+    when:
+      - not prometheus_docker_install
+      - prometheus_docker_compose_labels != []
+
+  - name: Fail if extra compose config is set but using docker isn't
+    debug:
+      msg: "Labels are set, but the role isn't configured to install prometheus with docker"
+    when:
+      - not prometheus_docker_install
+      - prometheus_docker_compose_extra_config != {}
+
+  - name: Get docker executable path
+    shell: whereis docker | awk '{print $2}'
```

Comment:
If needed, there should be a check for `dockerd` (daemon) and not `docker` (client).

Reply:
I should also propably use the stat module : https://docs.ansible.com/ansible/latest/modules/stat_module.html#stat-module